lib/fs.js

```diff
@@ -688,14 +688,19 @@ fs.renameSync = function(oldPath, newPath) {
 };
 
 fs.truncate = function(path, len, callback) {
-  if (typeof path === 'number') {
+  if (Number.isInteger(path) && path >= 0)
```

Comment:
@trevnorris I am okay with this being in major :-) Let's get this done right. Please suggest improvements as well :-)